spec-tests/src/test/java/org/candlepin/spec/pools/OneSubPoolPerStackSpecTest.java

```diff
@@ -0,0 +1,805 @@
+            .startDate(OffsetDateTime.now().minusDays(3L))
+            .endDate(OffsetDateTime.now().plusDays(6L))
+            .contractNumber(REGULAR_STACKED_WITH_DIFF_STACK_ID)
+            .accountNumber("")
+            .orderNumber(""));
+    }
+
+    private void createTargetPoolAndProducts(OwnerDTO owner, String stackingId) {
+        ContentDTO ownerContent1 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ContentDTO ownerContent2 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ProductDTO ownerDerivedEngProduct = ownerProductApi.createProductByOwner(
+            owner.getKey(), Products.random());
+        ownerProductApi.addBatchContent(owner.getKey(), ownerDerivedEngProduct.getId(),
+            Map.of(ownerContent1.getId(), true, ownerContent2.getId(), true));
+
+        ProductDTO ownerDerivedSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(new AttributeDTO().name("host_limited").value("true"))));
+
+        ProductDTO ownerSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(
+                new AttributeDTO().name("host_limited").value("true"),
+                new AttributeDTO().name("stacking_id").value(stackingId),
+                new AttributeDTO().name("virt_limit").value("unlimited"))));
+
+        ownerApi.createPool(owner.getKey(), Pools.random(ownerSku)
+            .stackId(stackingId)
+            .stacked(true)
+            .startDate(OffsetDateTime.now().minusSeconds(5L))
+            .derivedProductId(ownerDerivedSku.getId())
+            .derivedProvidedProducts(Set.of(
+                new ProvidedProductDTO()
+            .productId(ownerDerivedEngProduct.getId())
+            .productName(ownerDerivedEngProduct.getName()))));
+    }
+
+    private String getAttributeValue(String name, List<AttributeDTO> attributes) {
+        List<AttributeDTO> matches = attributes.stream()
+            .filter(x -> name.equals(x.getName()))
+            .collect(Collectors.toList());
+        if (matches.size() > 0) {
+            return matches.get(0).getValue();
+        }
+        return null;
+    }
+
+    private List<PoolDTO> getHostPools(TestData data, List<String> contracts) {
+        return data.getHostClient().pools().listPoolsByConsumer(data.getHost().getUuid()).stream()
+            .filter(item -> contracts.contains(item.getContractNumber()))
+            .sorted(Comparator.comparing(item -> contracts.indexOf(item.getContractNumber())))
+            .collect(Collectors.toList());
+    }
+
+    private EntitlementDTO bindHostPool(TestData data, List<PoolDTO> pools, String contract) {
+        PoolDTO pool = pools.stream()
+            .filter(x -> contract.equals(x.getContractNumber()))
+            .collect(Collectors.toList()).get(0);
```

Comment:
You may get an index out of bounds exception here if the list of pools doesn't have an element that has a contract number that matches the provided contract. You may need to check if the filtered list of pools is empty first and if it is, maybe return null or maybe do an assertion if you expect just one element to always be found.